nvtabular/column_group.py

```diff
@@ -179,27 +207,160 @@ def __getitem__(self, columns):
         """
         if isinstance(columns, str):
             columns = [columns]
+        if isinstance(columns, int):
+            return self.column_names[columns]
 
-        child = ColumnGroup(columns)
+        filtered_columns = [col for col in _convert_col(columns) if col.name in self.column_names]
+        child = ColumnGroup(filtered_columns)
         child.parents = [self]
         self.children.append(child)
         child.kind = str(columns)
         return child
 
+    def filter_columns(self, filter_fn, by_name=True):
+        if by_name:
+            filtered = [c for c in self.columns if filter_fn(c.name)]
+        else:
+            filtered = [c for c in self.columns if filter_fn(c)]
+
+        return self[filtered]
+
+    def filter_by_namespace(self, namespace):
+        return self.filter_columns(lambda c: c.startswith(namespace))
+
+    def get_tagged(self, tags, output_list=False, tags_to_filter=None):
+        column_names_to_filter = (
+            self.get_tagged(tags_to_filter, output_list=True) if tags_to_filter else []
+        )
+
+        if isinstance(tags, DefaultTags):
+            tags = tags.value
+        if not isinstance(tags, list):
+            tags = [tags]
+        output_cols = []
+
+        for column in self.flattened_columns:
+            if all(x in column.tags for x in tags):
+                output_cols.append(column)
+
+        columns = [col for col in output_cols if col.name not in column_names_to_filter]
+
+        if output_list:
+            return [col.name for col in columns]
+
+        child = ColumnGroup(columns, tags=tags)
+        child.parents = [self]
+        self.children.append(child)
+        # child.kind = f"tagged={child._tags_repr} " + self._cols_repr
+
+        return child
+
+    def tags_by_column(self):
+        outputs = {}
+
+        for col in self.flattened_columns:
+            outputs[col.name] = col.tags
+
+        return outputs
+
+    def targets_columns(self):
+        return self.get_tagged(Tag.TARGETS, output_list=True)
```

Comment:
If we can sort out a way to have `get_tagged` return a consistent type, could we live without these additional public methods? It seems like they're mostly syntactic sugar to avoid needing to pass the flag, and it would be hard to add methods like this for custom tags.